ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java

```diff
@@ -136,33 +136,33 @@ private static Map getElementDecls() {
             // Commonly used variables
             Set emptySet = new TreeSet();
             String spaceAtt = "xml:space";  // NOI18N
-            Set tmpSet;
+            Set<String> tmpSet;
 
             ////////////////////////////////////////////////
             // Initialize common sets
 
-            Set charInstructions = new TreeSet(Arrays.asList(new String[]{"apply-templates", // NOI18N
+            Set<String> charInstructions = new TreeSet<>(Arrays.<String>asList(new String[]{"apply-templates", // NOI18N
             "call-template","apply-imports","for-each","value-of", // NOI18N
             "copy-of","number","choose","if","text","copy", // NOI18N
             "variable","message","fallback"})); // NOI18N
 
-            Set instructions = new TreeSet(charInstructions);
-            instructions.addAll(Arrays.asList(new String[]{"processing-instruction", // NOI18N
+            Set<String> instructions = new TreeSet<>(charInstructions);
+            instructions.addAll(Arrays.<String>asList(new String[]{"processing-instruction", // NOI18N
```

Comment:
inferrable type informations
```suggestion
            instructions.addAll(Arrays.asList(new String[]{"processing-instruction", // NOI18N
```

strictly speaking the array creation is not needed as well

Reply:
changed..

Reply:
changed..